src/inference/src/dev/core_impl.cpp

```diff
@@ -204,7 +204,7 @@ ov::Plugin ov::CoreImpl::get_plugin(const std::string& pluginName) const {
             allowNotImplemented([&]() {
                 // Add device specific value to support device_name.device_id cases
                 std::vector<std::string> supportedConfigKeys =
-                    plugin.get_property(METRIC_KEY(SUPPORTED_CONFIG_KEYS), {});
+                    plugin.get_property(METRIC_KEY(SUPPORTED_CONFIG_KEYS), {}).as<std::vector<std::string>>();
```

Comment:
Confused, removing deprecated api will cause code become redundant.
Is it expected?

Reply:
We can use `auto supportedConfigKeys = ...` :)

Reply:
> We can use `auto supportedConfigKeys = ...` :)

If we change to `auto supportedConfigKeys = ...`, the type of 'supportedConfigKeys' is 'ov::Any'. The using of 'supportedConfigKeys' is inconvenient, such as the next line' supportedConfigKeys.begin()' no member 'begin()'

Reply:
> We can use `auto supportedConfigKeys = ...` :)

Got it! Will change